core/src/test/scala/cats/derived/show.scala

```diff
@@ -0,0 +1,31 @@
+package cats.derived
+import cats.Show
+import cats.derived.TestDefns.{Foo, Node, Tree, Leaf}
+import MkShow._
+
+class ShowTests extends KittensSuite {
+
+  test("for simple product") {
+    import cats.instances.all._
+    val foo = Foo(42, Option("Hello"))
+    assert(Show[Foo].show(foo) == "Foo(i = 42, b = Some(Hello))")
```

Comment:
@milessabin @LukaJCB what are you thoughts on this? I think @SystemFw's solution, although not perfect but a great addition to kittens. It would be nice merge and release kittens with some `Show` derivation.